Update peer dependency to allow React 17

**Describe the bug**  
Currently, the `package.json` in `8.2.0-beta2` requires React 16 as a peer dependency and is not compatible with React 17 (although it actually is of course). 

**To Reproduce**  
If you're trying to install `@react-native-mapbox-gl/map` together with **React 17** and **npm 7**, you will get a peer dependency error as follows:

```sh
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: myproject@1.0.0
npm ERR! Found: react@17.0.1
npm ERR! node_modules/react
npm ERR!   react@"17.0.1" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^16.6.1" from @react-native-mapbox-gl/maps@8.2.0-beta2
npm ERR! node_modules/@react-native-mapbox-gl/maps
npm ERR!   @react-native-mapbox-gl/maps@"^8.2.0-beta2" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /Users/arabold/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/arabold/.npm/_logs/2021-06-05T14_17_50_843Z-debug.log
```

This can be worked around by running `npm install --force` or `npm install --legacy-peer-deps`.

**Expected behavior** 
At a minimum the peer dependencies should be changed to allow any React version *larger or equal to 16.6.1*. That would include 17 and subsequent releases:

```json
  "peerDependencies": {
    "prop-types": ">=15.5.8",
    "react": ">=16.6.1",
    "react-native": ">=0.59.9"
  },
```

**Versions:**  
- `@mapbox/mapbox-sdk`: ^0.12.1
- `@react-native-mapbox-gl/maps`: ^8.2.0-beta2
- `react`: 17.0.1
- `react-dom`: 17.0.1
- `react-native`: ^0.64.1

--- a/package.json
+++ b/package.json
@@ -30,7 +30,7 @@
   },
   "peerDependencies": {
     "prop-types": ">=15.5.8",
-    "react": "^16.6.1",
+    "react": ">=16.6.1",
     "react-native": ">=0.59.9"
   },
   "dependencies": {
@@ -65,7 +65,7 @@
     "lint-staged": "^11.0.0",
     "metro-react-native-babel-preset": "0.49.1",
     "node-dir": "0.1.17",
-    "react": "16.8.6",
+    "react": "16.8.3",
     "react-docgen": "^5.0.0-beta.1",
     "react-native": "0.59.10",
     "typescript": "4.3.2",

